storage/innobase/include/arch0arch.h

```diff
@@ -0,0 +1,1984 @@
+  bool is_durable() const { return (m_dur_ref_count > 0); }
+
+  /** Purge archived files until the specified purge LSN.
+  @param[in]    purge_lsn       LSN until which archived files needs to be
+  purged
+  @param[out]   purged_lsn      LSN until which purging is successful;
+  LSN_MAX if there was no purging done
+  @return error code */
+  uint purge(lsn_t purge_lsn, lsn_t &purged_lsn);
+
+  /** Operations to be done at the time of shutdown. */
+  static void shutdown() { s_dblwr_file_ctx.close(); }
+
+  /** Update the reset information in the in-memory structure that we maintain
+  for faster access.
+  @param[in]    lsn     lsn at the time of reset
+  @param[in]    pos     pos at the time of reset */
+  void save_reset_point_in_mem(lsn_t lsn, Arch_Page_Pos pos)
+  {
+    m_file_ctx.save_reset_point_in_mem(lsn, pos);
+  }
+
+  /** Update stop lsn of a file in the group.
+  @param[in]    pos             stop position
+  @param[in]    stop_lsn        stop point */
+  void update_stop_point(Arch_Page_Pos pos, lsn_t stop_lsn)
+  {
+    m_file_ctx.update_stop_point(
+        Arch_Block::get_file_index(pos.m_block_num, ARCH_DATA_BLOCK), stop_lsn);
+  }
+
+  /** Recover the information belonging to this group from the archived files.
+  @param[in,out]        group_info      structure containing information of a
+  group obtained during recovery by scanning files
+  @param[in]      dblwr_ctx   file context related to doublewrite buffer
+  @return error code */
+  dberr_t recover(Arch_Recv_Group_Info &group_info, Arch_Dblwr_Ctx *dblwr_ctx);
+
+  /** Parse block for block info (header/data).
+  @param[in]    cur_pos         position to read
+  @param[in,out]        buff    buffer into which to write the parsed data
+  @param[in]    buff_len        length of the buffer
+  @return error code */
+  int read_data(Arch_Page_Pos cur_pos, byte *buff, uint buff_len);
+
+  /** Get archived file name at specific index in this group.
+  Caller would use it to open and copy data from archived files.
+  @param[in]    idx             file index in the group
+  @param[out]   name_buf        file name and path. Caller must
+                                  allocate the buffer.
+  @param[in]    buf_len         allocated buffer length */
+  void get_file_name(uint idx, char *name_buf, uint buf_len)
+  {
+    ut_ad(name_buf != nullptr);
+
+    /* Build name from the file context. */
+    m_file_ctx.build_name(idx, m_begin_lsn, name_buf, buf_len);
+  }
+
+  /** Get the current file size for this group.
+  Fixed size files are used for archiving data in a group.
+  @return file size in bytes */
+  uint64_t get_file_size() const { return m_file_size; }
+
+  /** Get aligned start LSN for this group
+  @return aligned start LSN */
+  lsn_t get_first_lsn() const { return (m_first_lsn); }
+
+  /** Get start LSN for this group
+  @return start LSN */
+  lsn_t get_begin_lsn() const { return (m_begin_lsn); }
```

Comment:
Unreachable. Do we need `m_begin_lsn` at all? I guess `log_sys.clone_lsn` would be better?